scripts/globals/magiantrials.lua

```diff
@@ -1,312 +1,1200 @@
+             end
+         end,
+
+  [ 892] = function (player, conditions)
+             if conditions.mob and conditions.mob:getID() == 17395816 then -- Moo Ouzi the Swiftblade
+                 return true
+             end
+         end,
+
+  [ 893] = function (player, conditions)
+             if conditions.mob and conditions.mob:getID() == 17223797 then -- Ni'Zho Bladebender
+                 return true
+             end
+         end,
+
+  [ 894] = function (player, conditions)
+             if conditions.mob and (conditions.mob:getID() == 16822423 or conditions.mob:getID() == 16822427) then -- Bugbear Strongman
+                 return true
+             end
+         end,
+
+  [ 895] = function (player, conditions)
+             if conditions.mob and conditions.mob:getID() == 17121576 then -- La Velue
+                 return true
+             end
+         end,
+
+  [ 896] = function (player, conditions)
+             if conditions.mob and conditions.mob:getID() == 17596628 then -- Hovering Hotpot
+                 return true
+             end
+         end,
+
+  [ 897] = function (player, conditions)
+             if conditions.mob and (conditions.mob:getID() == 17191325 or conditions.mob:getID() == 17109384 or conditions.mob:getID() == 17113491) then -- Yacumama
+                 return true
+             end
+         end,
+
+  [ 898] = function (player, conditions)
+             if conditions.mob and (conditions.mob:getID() == 17334552 or conditions.mob:getID() == 17338598) then -- Feuerunke
+                 return true
+             end
+         end,
+
+  [1758] = function (player, conditions)
+             if conditions.mob and (conditions.mob:getID() == 17195484) then -- Tammuz
+                 return true
+             end
+         end,
+}
+-------------------
+-- Magian Orange --
+-------------------
+
+function magianOrangeOnTrigger(player, npc)
+    local activeTrials = readTrials(player)
+    local p, t = parseParams(player)
+
+    if player:getMainLvl() < 75 then
+        player:startEvent(10121) -- can't take a trial before lvl 75
+
+    elseif (player:hasKeyItem(tpz.ki.MAGIAN_TRIAL_LOG) == false) then
+        player:startEvent(10122) -- player can start magian for the first time
+
+    else
+        player:startEvent(10123,p[1],p[2],p[3],p[4],p[5],0,0,t) -- standard dialogue
     end
 end
 
+-- set of non-weapon type items that can take trials
+local typeExceptions =
+{
+  [16191] = true,
+  [16192] = true,
+  [16193] = true,
+  [16194] = true,
+  [11926] = true,
+  [16199] = true,
+  [16200] = true,
+  [16198] = true,
+  [16197] = true,
+  [16196] = true,
+  [16195] = true,
+  [15070] = true,
+  [11927] = true,
+  [18573] = true,
+  [18571] = true,
+  [18574] = true,
+  [18575] = true,
+  [18576] = true,
+  [18839] = true,
+  [18342] = true,
+  [18572] = true,
+  [18577] = true,
+  [18578] = true,
+  [18579] = true,
+  [18580] = true,
+  [18840] = true
+}
+
+function magianOrangeOnTrade(player,npc,trade)
```

Comment:
we should avoid functions in scripts/globals luas that sit in the global space, to avoid clashes.  instead, look at how others do it -- for example scripts/globals/treasure.lua.  So something like

```lua
tpz = tpz or {}
tpz.magian = tpz.magian or {}

.
.
.

function tpz.magian.magianOrangeOnTrade(player, npc, trade)
.
.
.
end
```

Reply:
I gotchu Wren